Application/Migrator/SQLToMongoMigrator.cs

```diff
@@ -0,0 +1,133 @@
+
+            var name_entry_geo_location = connection.Query<name_entry_geo_location>("select name_entry_id, geo_location_place from name_entry_geo_location");
+
+            var geolocationPlace = from a in geolocation
+                                   join b in name_entry_geo_location
+                                   on a.place equals b.geo_location_place
+                                   select new { a.region, a.place, b.name_entry_id };
+
+
+            var name_entry = connection.Query<nameentry>("select id, created_at, extended_meaning, famous_people,ipa_notation, is_indexed, meaning, media, morphology, pronunciation, submitted_by, syllables, updated_at, variants, name, geo_location_id, state from name_entry");
+
+            if(name_entry == null)  return "No data found in MySQL table.";
+
+        
+            foreach (var item in name_entry)
+            {
+                item.etymology = etymology.Where(f => f.name_entry_id == item.id).ToList()
+                    .Select(s => new Etymology(s.part, s.meaning) { }).ToList();
+
+                item.geoLocations = geolocationPlace.Where(d=>d.name_entry_id == item.id).
+                    Select(a=> new GeoLocation() { Place =a.place, Region = a.region}).ToList();
+            }
+          
+            List<NameEntry> documentsToInsert = name_entry.Select(s => new NameEntry()
+            {
+                Id = ObjectId.GenerateNewId().ToString(),
+                Name = s.name,
+                //CreatedAt = ((DateTime)s.created_at,
+                ExtendedMeaning = s.extended_meaning,
+                FamousPeople = s.famous_people.Split(",").ToList(),
+                IpaNotation = s.ipa_notation,
+                Meaning = s.meaning,
+                Media = s.media.Split(",").ToList(),
+                Morphology = s.morphology.Split(",").ToList(),
+                Pronunciation = s.pronunciation,
+                CreatedBy = s.submitted_by,
+                Syllables = s.syllables.Split(",").ToList(),
+                //UpdatedAt = (DateTime)s.updated_at,
+                Variants = s.variants.Split(",").ToList(),
+                State = GetPublishState(s.state),
+                Etymology = s.etymology,
+                GeoLocation = s.geoLocations
+            }).ToList();
+        
+            _nameEntryCollection.DeleteMany(FilterDefinition<NameEntry>.Empty);
+            _nameEntryCollection.InsertMany(documentsToInsert);
+            return $"{documentsToInsert.Count} name records inserted into MongoDB successfully.";
+           
+        }
+
+
+        private State GetPublishState(string input)
+        {
+            switch (input)
+            {
+                case "PUBLISHED": return State.PUBLISHED;
+                case "NEW": return State.NEW;
+                case "UNPUBLISHED": return State.UNPUBLISHED;
+                case "MODIFIED": return State.MODIFIED;
+
+                default:
+                    return State.PUBLISHED;
```

Comment:
Names without a status should have status NEW. Other than that, looks good to me. Feel free to merge in after making that change.